### ISSUE TITLE
Last legend line is cut when navigation is enabled and symbolHeight is set

### ISSUE THREAD

**Issue by pawelfus** [highcharts/highcharts#12282]
#### Expected behaviour
Nice round number of lines in legend when navigation is on:
![Screen Shot 2019-10-28 at 14 21 34](https://user-images.githubusercontent.com/1453926/67681792-4f24e100-f98e-11e9-823d-5a2293f03cf7.png)

#### Actual behaviour
Last line is cut 
![Screen Shot 2019-10-28 at 14 21 43](https://user-images.githubusercontent.com/1453926/67681794-4f24e100-f98e-11e9-970c-5e80952c2f3d.png)

#### Live demo with steps to reproduce
http://jsfiddle.net/BlackLabel/xv3ck920/
vs 
http://jsfiddle.net/BlackLabel/xv3ck920/2/

#### Workaround
Demo: http://jsfiddle.net/BlackLabel/jnaw6x0s/

Plugin:
```js

  (function(H) {
    H.wrap(
      H.Legend.prototype,
      'handleOverflow',
      function(proceed, legendHeight) {
        var chart = this.chart,
          options = this.options,
          maxHeight = options.maxHeight,
          itemSize = options.itemMarginTop + options.symbolHeight,
          spaceHeight = (
            chart.spacingBox.height +
            (options.verticalAlign === 'top' ? -options.y : options.y) -
            this.padding
          );


        if (!H.defined(maxHeight)) {
          if (
            options.layout === 'horizontal' &&
            options.verticalAlign !== 'middle' &&
            !options.floating
          ) {
            spaceHeight /= 2;
          }

          // Rounding
          spaceHeight -= spaceHeight % itemSize + this.padding / 2;

          this.options.maxHeight = spaceHeight;
        }
        legendHeight = proceed.apply(
          this,
          Array.prototype.slice.call(arguments, 1)
        );

        this.options.maxHeight = maxHeight;
        return legendHeight;
      }
    );
  })(Highcharts);
```

**Comment by VadimDyliko** [comment]
Hi @pawelfus 
I have tested your workaround. This solution does not work with certain heights and count of entry's: 
![image](https://user-images.githubusercontent.com/44729046/67688034-417d5480-f9aa-11e9-9b9a-0387927f50d2.png)

**Comment by VadimDyliko** [comment]
@pawelfus, yes, of course. It happens if set the height of the container to 500px, for example ( http://jsfiddle.net/VadimDyliko/yvLj1xz5/ ) or set more entry's ( http://jsfiddle.net/VadimDyliko/yvLj1xz5/3/ )

**Comment by pawelfus** [comment]
Thanks! It seems like a bit more complicated issue than I thought. I was able to temporary resolve the issue for your case: http://jsfiddle.net/BlackLabel/7xzk60ja/ but increasing drastically `symbolHeight` is not solved. If setting fixed `maxHeight` or workaround above won't resolve the issue, then I suggest to disable `symbolHeight` and `itemMarginTop` or keep labels cut off.

*Internal note:*
In general it's a problem how we apply options like `itemMarginTop/Bottom` - not to the legend item, but to the legend text. And when we calculate item's height, we pick first available: text's bbox or `symbolHeight` option, while picking the highest value should be considered: http://jsfiddle.net/BlackLabel/6f8mxkjz/3/ (fixed `legendItemHeight`).

**Comment by VadimDyliko** [comment]
Unfortunately, none of the proposed solutions to the problem is suitable for us. Let me inform you, that the code work well with out this condition at string 22997 of highcharts.js file.
![image](https://user-images.githubusercontent.com/44729046/67839649-2d019f00-fb05-11e9-95e6-a43c7f1e2957.png)

**Comment by emanette** [comment]
We use Highcharts at Envestnet Tamarac and I'm having this issue of the last line being cut off in the legend by the navigation bar. I have a maxHeight set for the legend. Is there a fix coming soon?

**Comment by pawelfus** [comment]
*Internal note:*
Make sure that auto-calculated `maxHeight` takes into account all options described in the issue.

**Comment by emanette** [comment]
Are you referring to `symbolHeight `and `itemMarginTop`? We don't have either of those set, and I've set the `maxHeight `to a % of the chart height.

**Comment by pawelfus** [comment]
Thank you for the details. For example, if you set `maxHeight` to 93px and each line requires 20px, then the last one will be cut off. This issue is about setting `symbolHeight` and `itemMarginTop` which miscalculates auto-`maxHeight`. It seems your issue is different. If you think this is a bug, please create a new report, so we will discuss this there.
